### ISSUE TITLE
Prepare repository for R24-05

### ISSUE THREAD

**Issue by arnoweiss** [eclipse-tractusx/tractus-x-umbrella#87]
### Acceptance Criteria

- [ ] tractusx-edc bumped to 0.7.x
- [ ] communication between wallet and tractusx-edc via IATP VPP
- [ ] configuration payloads adjusted according to R24-05 standards and kits

### Additional Information
This is an umbrella feature-request that probably can't be satisfied as long as there's no FOSS implementation of an IATP Credential Service. However, it may make sense to plan ahead to see what's missing until CX Release 24-05 can be satisfied and plan child issues to this one. If this is the right place to plan this, feel free to involve me here. Else, feel free to close.

**Comment by evegufy** [comment]
Hi @arnoweiss a first step in the direction of R24-05 was done with https://github.com/eclipse-tractusx/tractus-x-umbrella/pull/94, at least preparing the Rights and Roles Setup (CX-Central Realm, Keycloak). Still, change to R24-05 is challenging because be don't have a FOSS wallet (yet, AFAIK MIW is trying to catch up with the implementation).
@tom-rm-meyer-ISST I might be wrong, but did I spot a mock for the DIM in your product (PURIS)? And if so, do you think it would be suitable to using it in the umbrella?

**Comment by tom-rm-meyer-ISST** [comment]
> Hi @arnoweiss a first step in the direction of R24-05 was done with #94, at least preparing the Rights and Roles Setup (CX-Central Realm, Keycloak). Still, change to R24-05 is challenging because be don't have a FOSS wallet (yet, AFAIK MIW is trying to catch up with the implementation). @tom-rm-meyer-ISST I might be wrong, but did I spot a mock for the DIM in your product (PURIS)? And if so, do you think it would be suitable to using it in the umbrella?

@evegufy exactly. It's located [here](https://github.com/eclipse-tractusx/puris/tree/main/local/iam-mock) and coded in pyhton.

It's containerized but not published. Beside that we did not create a helm chart and did not make things configurable as it should just bridge till R24.08 - at least I [documented](https://github.com/eclipse-tractusx/puris/tree/main/local/iam-mock) what I learned regarding the needed [components](https://github.com/eclipse-tractusx/puris/blob/main/local/iam-mock/docs/README_SERVICE_OVERVIEW.md).
- in the generate-keys.sh we create 
  - [mock key pair used for all dids](https://github.com/eclipse-tractusx/puris/blob/main/local/generate-keys.sh#L125) -> likely we can use something from portal here? 
  - [private keys to sign jwt](https://github.com/eclipse-tractusx/puris/blob/main/local/generate-keys.sh#L129), [public keys to allow the edc to verify](https://github.com/eclipse-tractusx/puris/blob/main/local/generate-keys.sh#L105)
  - [the seeding script for the bdrs](https://github.com/eclipse-tractusx/puris/blob/main/local/generate-keys.sh#L133)
- we hard coded the [roles](https://github.com/eclipse-tractusx/puris/blob/main/local/iam-mock/constants.py#L41) as we just need the issuer and two participants
- Configuration of an EDC is [here](https://github.com/eclipse-tractusx/puris/blob/main/local/tractus-x-edc/config/customer/control-plane.properties#L21).
  - We still need a valid keycloak to be hosted - we don't have a dependency here directly, but the EDC will hand over bearer tokens
  - we need a vault to store some secrets (public or private key, would need to check the detail)

So basically - beside creating the helm chart - one could enhance it a little or lift, shift and update the hard coded roles accordingly :) The mock does NOT offer the storage api to hand over credentials.

**Comment by evegufy** [comment]
Upgrade to R24-05 did happen. With some limitations, see https://github.com/eclipse-tractusx/tractus-x-umbrella/tree/umbrella-1.0.0/charts/umbrella#note-for-r2405, which hopefully can be addressed with https://github.com/eclipse-tractusx/tractus-x-umbrella/issues/122

**Comment by evegufy** [comment]
@arnoweiss IMO this issue can be closed
